Bezier.cpp

```diff
@@ -4,9 +4,70 @@
 Bezier::Bezier(const Vector2& p0, const Vector2& p1, const Vector2& p2, const Vector2& p3)
 	: p0(p0), p1(p1), p2(p2), p3(p3)
 {
+	// Assign points to Bezier
+	this->p0 = p0;
+	this->p1 = p1;
+	this->p2 = p2;
+	this->p3 = p3;
+
+	calculatePoints();
+}
+
+void Bezier::calculatePoints()
+{
+	// Generate curves based on desired smoothness
+	float t = 0;
+	// Angle of line
+	float angle = 0;
+
+	// Generate array of SDL_Point to use in rendering lines
+	for (int i = 0; i < CURVES + 1; i++)
+	{
+		points[i] = Vec2toSDLPoint(getPosAlongBezier(t));
+
+		// Calculate the 
+		if (i > 0)
+		{
+			// Calculate angles for all but the first line of the curve
+			angle = atan2(points[i].y - points[i - 1].y, points[i].x - points[i - 1].x);
+
+			// Calculate corresponding outer and inner points, offset by half of the curve width each
+			pointsOuter[i].x = (WIDTH / 2) * sin(angle) + points[i].x;
+			pointsOuter[i].y = -(WIDTH / 2) * cos(angle) + points[i].y;
+			pointsInner[i].x = -(WIDTH / 2) * sin(angle) + points[i].x;
+			pointsInner[i].y = (WIDTH / 2) * cos(angle) + points[i].y;
+		}
+
+		t += 1.0 / CURVES;
```

Comment:
Incrementing `t` like this could lead to accumulating floating point errors -- it would be better to calculate `t = (float)i / CURVES` each time around the loop.